### PR TITLE
Adding GraphRequestBody for building multipart form data

### PR DESCRIPTION
**Pull request by joesus**
## Checklist

- [x] I've read the [Contributing Guidelines](CONTRIBUTING.md) and the [Code of Conduct](CODE_OF_CONDUCT.md)
- [x] I've completed the [Contributor License Agreement](https://developers.facebook.com/opensource/cla)
- [x] I've ensured that all existing tests pass and added tests (when/where necessary)
- [x] I've ensured that my code lints properly: (`swiftlint && swiftlint autocorrect --format`)
- [x] I've updated the documentation (when/where necessary) and [Changelog](CHANGELOG.md) (when/where necessary)
- [x] I've added the proper label to this pull request (e.g. `bug` for bug fixes)

## Pull Request Details

Introduces `GraphRequestBody` for making POST requests.

## Test Plan

Unit tests should cover the new functionality

### CODE REVIEW

File: FacebookSwift.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -2340,7 +2353,7 @@
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks @loader_path/Frameworks";
 				PRODUCT_BUNDLE_IDENTIFIER = "--PRODUCT-BUNDLE-IDENTIFIER-";
 				SWIFT_OBJC_BRIDGING_HEADER = "$(PROJECT_DIR)/Sources/Core/Internal/FacebookSwift-Bridging-Header.h";
-				SWIFT_VERSION = 4.2;
+				SWIFT_VERSION = 5.0;
```
Comment (jingping2015) [inline FacebookSwift.xcodeproj/project.pbxproj]: what's this version? is this one sync with objC version ?

Comment (joesus) [reply FacebookSwift.xcodeproj/project.pbxproj]: This was because I missed a checkbox earlier on one of the targets. It's boilerplate. :)

**Comment by dreamolight** [comment]
I think we should use multipart form when there are both data and json? As `application/json` can't handle binary data (without encoding).

**Comment by joesus** [comment]
> I think we should use multipart form when there are both data and json? As `application/json` can't handle binary data (without encoding).

That's what this does. I'll double back and make sure there's a test case for this but basically I just used the original implementation where it keeps a running total of data regardless of what type is appended. If something other than json is appended then it clears out the attached json... There's a non-zero chance that there are bugs in the original as well. I'll try to address both. :D

**Comment by joesus** [comment]
> I think we should use multipart form when there are both data and json? As `application/json` can't handle binary data (without encoding).

Looked again. Added flag for tracking whether the multipart form data format should be used. I thought it would be easier to read and require less domain knowledge for the next person who works on it. I didn't like the idea of bringing extra state into it but it seemed preferable to using an optional collection's existence as a signifier. Lemme know if you think this is easier to reason about or if I should just use the optional collection approach. Your call.

**Pull request merged**